ctx.py

```diff
@@ -13,22 +13,104 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import enum
 import functools
+import os
 
+from pathlib import Path
+
+import util
+
+from concourse.pipelines.modelbase import ModelBase
 '''
 Execution context. Filled upon invocation of cli.py, read by submodules
 '''
 
 args=None # the parsed command line arguments
 
 
-def _cfg_factory_from_dir():
-    # XXX: args does always have a cfg_dir attribute, but pylint does not always understand this
-    if not args or not hasattr(args, 'cfg_dir') or not getattr(args, 'cfg_dir'):
-        return None
+class Config(ModelBase):
+
+    def __init__(self,):
+        super().__init__(raw_dict={})
+
+    def _update(self, update_dict: dict):
+        self.raw = util.merge_dicts(self.raw, update_dict)
+
+    def _config_key(self):
+        pass
+
+    def _load_home_config_file(self):
+        config_file = Path.home() / '.cc-utils.cfg'
+        if config_file.exists() and config_file.is_file():
+            return util.parse_yaml_file(config_file)
+        else:
+            return None
+
+    def _config_from_home(self):
+        config_file_contents = self._load_home_config_file()
+        if config_file_contents is None:
+            return {}
+
+        return config_file_contents.get(self._config_key(), {})
+
+    def _config_from_env(self):
+        return {}
```

Comment:
this will be moved out; but: if it would not, this should raise NotImpementedError